internal/build/promote_test.go

```diff
@@ -58,11 +58,28 @@ func TestPromote(t *testing.T) {
 	)
 }
 
+func TestPromoteWithInvalidVersion(t *testing.T) {
+	opts := Opts{
+		Environment: "testing",
+		Project:     "promote-build",
+		Version:     "1.1.2-updated",
+	}
+
+	_, err := setup(setupOpts{
+		env:     opts.Environment,
+		project: opts.Project,
+	})
+	assert.Nil(t, err)
```

Comment:
Prefer `require` over `assert`.

`require` stops the test if the condition is not met. `assert` records the error but continues.
For conditions that render the test invalid such as failed setup -> nothing can go right after that.

Even in conditions where assert is justified, the noise it creates is generally a headache.